ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template4.kt

```diff
@@ -0,0 +1,383 @@
+        verticalArrangement = Arrangement.spacedBy(Template4UIConstants.packageButtonContentVerticalSpacing),
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        OfferName(
+            packageInfo = packageInfo,
+            textColor = colors.text1,
+            modifier = Modifier
+                .wrapContentSize()
+                .weight(1f),
+        )
+
+        Text(
+            text = packageInfo.rcPackage.product.price.formatted,
+            color = colors.text1,
+            style = MaterialTheme.typography.bodyLarge,
+            fontWeight = FontWeight.SemiBold,
+        )
+    }
+}
+
+@Composable
+private fun DiscountRelativeToMostExpensivePerMonth(
+    packageInfo: TemplateConfiguration.PackageInfo,
+    colors: TemplateConfiguration.Colors,
+) {
+    val applicationContext = LocalContext.current.applicationContext.toAndroidContext()
+    val text = packageInfo.localizedDiscount(applicationContext).uppercase()
+    AutoResizedText(
+        text = text,
+        color = colors.text1,
+        style = MaterialTheme.typography.bodySmall,
+        fontWeight = FontWeight.Bold,
+        textAlign = TextAlign.Center,
+        modifier = Modifier.padding(
+            horizontal = UIConstant.defaultHorizontalPadding,
+            vertical = Template4UIConstants.discountVerticalPadding,
+        ),
+    )
+}
+
+@Composable
+private fun OfferName(
+    packageInfo: TemplateConfiguration.PackageInfo,
+    textColor: Color,
+    modifier: Modifier = Modifier,
+) {
+    var firstRow: String? = null
+    var secondRow: String = packageInfo.rcPackage.product.title
+
+    packageInfo.localization.offerName?.let { offerName ->
+        val components = offerName.split(" ", limit = 2)
+        if (components.size == 2) {
+            firstRow = components[0]
+            secondRow = components[1]
+        } else {
+            secondRow = offerName
+        }
+    }
+
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = modifier,
+    ) {
+        firstRow?.let {
+            Text(
+                text = it,
+                color = textColor,
+                style = MaterialTheme.typography.titleLarge,
+                fontWeight = FontWeight.Bold,
+                textAlign = TextAlign.Center,
+            )
+        }
+
+        Text(
+            text = secondRow,
+            color = textColor,
+            style = MaterialTheme.typography.bodyLarge,
+            fontWeight = FontWeight.Normal,
+            textAlign = TextAlign.Center,
+        )
+    }
+}
+
+@Composable
+private fun CheckmarkBox(
+    isSelected: Boolean,
+    colors: TemplateConfiguration.Colors,
+    modifier: Modifier = Modifier,
+) {
+    Box(
+        modifier = modifier
+            .size(Template4UIConstants.checkmarkSize)
+            .clip(CircleShape),
+    ) {
+        if (isSelected) {
+            PaywallIcon(icon = PaywallIconName.CHECK_CIRCLE, tintColor = colors.accent1)
+        }
```

Comment:
Let's use `AnimatedVisibility` here instead so that it fades in and out?

Reply:
done in [bb4d92b](https://github.com/RevenueCat/purchases-android/pull/1349/commits/bb4d92bda032667e8dbb075ddafbc92fd8c94471)

I used same parameters as the PurchaseButton loading animation. It doesn't show that well in debug but I think it's because the performance is lower in debug builds?